solr/core/src/java/org/apache/solr/metrics/prometheus/SolrPrometheusFormatter.java

```diff
@@ -220,6 +281,11 @@ public MetricSnapshots collect() {
       snapshots.add(
           new GaugeSnapshot(new MetricMetadata(metricName), metricGauges.get(metricName)));
     }
+    for (String metricName : metricSummaries.keySet()) {
```

Comment:
Always iterate the entrySet or call forEach so you don't have to lookup the value of each one separately